gc/base/SparseAddressOrderedFixedSizeDataPool.hpp

```diff
@@ -0,0 +1,281 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *
+******************************************************************************/
+
+#if !defined(SparseAddressOrderedFixedSizeDataPool_HPP_)
+#define SparseAddressOrderedFixedSizeDataPool_HPP_
+
+#include "omrpool.h"
+#include "BaseVirtual.hpp"
+#include "EnvironmentBase.hpp"
+#include "GCExtensionsBase.hpp"
+#include "SparseHeapLinkedFreeHeader.hpp"
+
+/**
+ *
+ * @ingroup GC_Base_Core
+ */
+
+typedef struct SparseDataTableEntry {
+	void *dataPtr; /**< Object data pointer related to proxy object */
+	void *proxyObjPtr; /**< Pointer to proxy object that is residing in-heap */
+	uintptr_t size; /**< Total size of the data pointed to by dataPtr */
+#if defined(OMR_GC_DOUBLE_MAPPING_FOR_SPARSE_HEAP_ALLOCATION)
+	struct J9PortVmemIdentifier *identifier; /**< Identifier associated with double mapped region */
+#endif /* OMR_GC_DOUBLE_MAPPING_FOR_SPARSE_HEAP_ALLOCATION */
+
+	SparseDataTableEntry() {
+		dataPtr = NULL;
+		proxyObjPtr = NULL;
+		size = 0;
+#if defined(OMR_GC_DOUBLE_MAPPING_FOR_SPARSE_HEAP_ALLOCATION)
+		identifier = NULL;
+#endif /* OMR_GC_DOUBLE_MAPPING_FOR_SPARSE_HEAP_ALLOCATION */
+	}
+
+	SparseDataTableEntry(void *dataPtr_) {
+		dataPtr = dataPtr_;
+		proxyObjPtr = NULL;
+		size = 0;
+#if defined(OMR_GC_DOUBLE_MAPPING_FOR_SPARSE_HEAP_ALLOCATION)
+		identifier = NULL;
+#endif /* OMR_GC_DOUBLE_MAPPING_FOR_SPARSE_HEAP_ALLOCATION */
+	}
+
+	SparseDataTableEntry(void *dataPtr_, void* proxyObjPtr_, uintptr_t size_) {
+		dataPtr = dataPtr_;
+		proxyObjPtr = proxyObjPtr_;
+		size = size_;
```

Comment:
identifier = NULL?

btw, couldn't the 3 above be just one, using default values for params?

Reply:
Yes I will modify this, thank you. As for `identifier`, would you rather I just declared it instead of initializing it as well?